File: C++/Swap Nodes in Pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *swapPairs(ListNode *head) {
        ListNode *first = head;
        ListNode *next = NULL;
        ListNode *last = NULL;
        if (first != NULL) {
            next = first -> next;
            if (next != NULL) {
                head = next;
                last = first;
                first = next -> next;
                next -> next = last;
                last -> next = NULL;
            } else
                return head;
        }
        while (first != NULL) {
            next = first -> next;
            if (next != NULL) {
                last -> next = next;
                last = first;
                first = next -> next;
                next -> next = last;
                last -> next = NULL;
            } else {
                last -> next = first;
                first = NULL;
            }
        }
        return head;
    }
};

File: Java/Swap Nodes in Pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
	public ListNode swapPairs(ListNode head) {
		ListNode node = head;
		if (head != null && head.next != null)
			head = head.next;
		while (node != null && node.next != null) {
			ListNode next = node.next;
			node.next = next.next;
			next.next = node;
			if (node.next != null && node.next.next != null) {
				ListNode temp = node.next;
				node.next = node.next.next;
				node = temp;
			} else
				node = node.next;
		}
		return head;
	}
}